SampleScripts/PowerManagement-ManagementDomain.ps1

```diff
@@ -1032,7 +1205,145 @@ if ($PsBoundParameters.ContainsKey("startup")) {
+                                    Exit
+                                }
+
+                                #Start workflow for VCF prior version 4.5
+                                if ([float]$vcfVersion -lt [float]4.5) {
+                                    # Start vSphere HA
+                                    if (!$(Set-VsphereHA -server $vcfVcenterDetails.fqdn -user $vcfVcenterDetails.ssoAdmin -pass $vcfVcenterDetails.ssoAdminPass -cluster $clustername -enableHA)) {
+                                        Write-PowerManagementLogMessage -Type ERROR -Message "Could not enable vSphere High Availability for cluster '$cluster'."
+                                    }
+
+                                    # Restore the DRS Automation Level to the mode backed up for Management Domain Cluster during shutdown
+                                    if ([string]::IsNullOrEmpty($DrsAutomationLevel)) {
+                                        Write-PowerManagementLogMessage -Type ERROR -Message "The DrsAutomationLevel value in the JSON file is empty. Exiting!"
+                                        Exit
+                                    }
+                                    else {
+                                        Set-DrsAutomationLevel -server $vcfVcenterDetails.fqdn -user $vcfVcenterDetails.ssoAdmin -pass $vcfVcenterDetails.ssoAdminPass -cluster $clustername -level $DrsAutomationLevel
+                                    }
+                                }
 
+                                # Startup the vSphere Cluster Services Virtual Machines in the Management Workload Domain
+                                Set-Retreatmode -server $vcfVcenterDetails.fqdn -user $vcfVcenterDetails.ssoAdmin -pass $vcfVcenterDetails.ssoAdminPass -cluster $clustername -mode disable
+                                # Waiting for vCLS VMs to be started for ($retries*10) seconds
+                                Write-PowerManagementLogMessage -Type INFO -Message "vCLS retreat mode has been set. vCLS virtual machines startup will take some time. Please wait."
+                                $counter = 0
+                                $retries = 10
+                                $sleepTime = 30
+                                while ($counter -ne $retries) {
+                                    $powerOnVMcount = (Get-VMsWithPowerStatus -powerstate "poweredon"  -server $vcfVcenterDetails.fqdn -user $vcfVcenterDetails.ssoAdmin -pass $vcfVcenterDetails.ssoAdminPass -pattern "(^vCLS-\w{8}-\w{4}-\w{4}-\w{4}-\w{12})|(^vCLS\s*\(\d+\))|(^vCLS\s*$)" -silence).count
+                                    if ( $powerOnVMcount -lt 3 ) {
+                                        Write-PowerManagementLogMessage -Type INFO -Message "There are $powerOnVMcount vCLS virtual machines running. Sleeping for $sleepTime seconds until the next check."
```

Comment:
Should this be looking for powered off instead?

Reply:
This is a copy of the existing code, i will update once we start clean up